Maple2.Server.Game/PacketHandlers/FishingHandler.cs

```diff
@@ -0,0 +1,377 @@
+    public override RecvOp OpCode => RecvOp.Fishing;
+
+    private enum Command : byte {
+        Prepare = 0,
+        Stop = 1,
+        Catch = 8,
+        Start = 9,
+        FailMinigame = 10,
+    }
+
+    #region Autofac Autowired
+    // ReSharper disable MemberCanBePrivate.Global
+    public required ItemMetadataStorage ItemMetadata { private get; init; }
+    public required TableMetadataStorage TableMetadata { private get; init; }
+    // ReSharper restore All
+    #endregion
+
+    public override void Handle(GameSession session, IByteReader packet) {
+        var command = packet.Read<Command>();
+        switch (command) {
+            case Command.Prepare:
+                HandlePrepare(session, packet);
+                break;
+            case Command.Stop:
+                HandleStop(session);
+                break;
+            case Command.Catch:
+                HandleCatch(session, packet);
+                break;
+            case Command.Start:
+                HandleStart(session, packet);
+                break;
+            case Command.FailMinigame:
+                HandleFailMinigame(session);
+                break;
+        }
+    }
+
+    private void HandlePrepare(GameSession session, IByteReader packet) {
+        long fishingRodUid = packet.ReadLong();
+
+        if (session.Field == null || session.GuideObject != null) {
+            return;
+        }
+
+        Item? rod = session.Item.Inventory.Get(fishingRodUid);
+        if (rod == null || rod.Metadata.Function?.Type != ItemFunction.FishingRod) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_invalid_item));
+            return;
+        }
+
+        if (!int.TryParse(rod.Metadata.Function?.Parameters, out int rodCode) || !TableMetadata.FishingRodTable.Entries.TryGetValue(rodCode, out FishingRodTable.Entry? rodMetadata)) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_invalid_item));
+            return;
+        }
+
+        if (session.Mastery[MasteryType.Fishing] < rodMetadata.MinMastery) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_fishingrod_mastery));
+            return;
+        }
+
+        if (!TableMetadata.FishingSpotTable.Entries.TryGetValue(session.Player.Value.Character.MapId, out FishingSpotTable.Entry? spotMetadata)) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_notexist_fish));
+            return;
+        }
+
+        if (session.Mastery[MasteryType.Fishing] < spotMetadata.MinMastery) {
+            session.Send(FishingPacket.Error(FishingError.s_fishing_error_lack_mastery));
+            return;
+        }
+
+        IList<Vector3> fishingBlocks = GetFishingBlocks(session.Player.Position, session.Player.Rotation);
+
+        session.GuideObject = session.Field?.SpawnGuideObject(session.Player, new FishingGuideObject(), GetGuideObjectPosition(fishingBlocks, session.Player.Position));
```

Comment:
`session.Field` is guaranteed to not be null here, so you don't need null coalescing, and therefore you also don't need the null ignore on `GuideObject` below